### ISSUE TITLE
Make Momostr opt-in on Nostr side

### ISSUE THREAD

**Issue by nanikamado** [nanikamado/momostr#10]
# TL;DR
I'm planning to make Momostr opt-in on Nostr side.

# Motivation
Momostr is currently opt-out and not opt-in on both the Nostr and Fediverse sides, which means we do not get explicit consent from the account owners on Nostr or Fediveres to bridge their accounts.

This is causing the following problems:

### Duplicate Accounts with Multiple Nostr-Fediverse Bridges
If a Nostr user is followed by Fediverse users through two different instances of Nostr-Fediverse bridges, those bridges will create two copies of the same Nostr account separately. This is confusing.

Although one can opt out from a certain Momostr instance by using the bot, it's not great to manually opt out from each instance every time a new instance pops up.

Additionally, this issue causes people to avoid self-hosting their own Momostr instances to prevent confusion and this is not good. There should be multiple instances owned by various kinds of people with different moderation policies.

### Confusion Between Nostr and Fediverse Users
When I started running Momostr on momostr.pink, I was sending the bridged events to major Nostr relays like relay.nostr.band or relay.primal.net. This caused users of clients that do not interpret the proxy tag to mistake Fediverse users for Nostr users.

Therefore, I stopped using other relays and began sending bridged events only to relay.momostr.pink. I also added the `-` tag to prevent broadcasting by random people.

However, there are still some chances of confusion because clients can connect to relay.momostr.pink event if the user did not explicitly added it to the relay list, and random people still can broadcast events to other relays and relays can ignore the `-` tag.

### Difficulty in Preventing Spam
If there is a opt-in/account-registration step, we can use CAPTCHA and it will be much easier to filter out spam.

# What I'm Planning to Change
Making it opt-in only on Nostr side will solve all the problems described above, so I decided to require account registration or an opt-in step for Nostr users to use Momostr.

Additionally, instead of using npub for usernames, we can also ask users what username do they want to use for their bridge accounts on the registration step.

However, I'm not planning to make momostr.pink opt-in anytime soon as it is already used by many people, and making it opt-in suddenly might cause confusion. Instead, I will host a new Momostr instance on a different domain and see if things work well, while keeping momostr.pink as it is for the time being.

# What I Need to Do / Implement
- Create a webpage for the account registration.
- Configure the relay to require AUTH in order to read and write and block all the events from not-authed or unregistered accounts.
- Make it possible to set usernames of the bridge accounts.
